WordPress/Classes/ViewRelated/Reader/Detail/Views/ReaderDetailNewHeaderView.swift

```diff
@@ -60,8 +60,11 @@ class ReaderDetailNewHeaderViewHost: UIView {
         guard let swiftUIView = subviews.first else {
             return
         }
-        swiftUIView.invalidateIntrinsicContentSize()
-        layoutIfNeeded()
+
+        DispatchQueue.main.async {
+            swiftUIView.invalidateIntrinsicContentSize()
+            self.layoutIfNeeded()
+        }
```

Comment:
Did you consider annotating the method with `@MainActor`?

Reply:
Sorry for missing your comment, and thanks for the suggestion @mokagio! Annotating this method with `MainActor` is a great idea since I need this method to always run on the main thread. 

Turns out #21797 resurfaced, so I'll need to investigate this again. When the fix is up, I'll make sure to implement your suggestion there. 🙂